app/rfid.py

```diff
@@ -11,21 +17,17 @@ def __init__(self):
             parity=serial.PARITY_NONE,
             stopbits=serial.STOPBITS_ONE,
             bytesize=serial.EIGHTBITS,
-            timeout=0,
+            timeout=10
         )
-        # ser.write(b'SD2\r\n')
+        # ser.write(b'SD2\r\n') # Sets transponder default to FDX-B(animal tag)
 
     def read(self):
-        # ser.reset_input_buffer()
+        ser.reset_input_buffer()
         ser.reset_output_buffer()
-        ser.write(b'rat\r\n')
-        a = ser.read(34).decode('utf-8')
-        if '?1' in a:
+        ser.write(b'sd2\r\n')
+        a = ser.read_until(size=19).decode('utf-8')
```

Comment:
Why `19` - can we have a comment here?

I guess the relevant documentation is [here](http://www.priority1design.com.au/fdx-b_animal_identification_protocol.html)?  If I understand correctly, reading 19 bytes (152 bits) is more than the 128 bits defined in the FDX-B spec which means this would block.  Which I presume it doesn't as you've tested it... but that's why this could do with comments to explain the magic numbers.

Reply:
Done